### PR TITLE
Craft access token

### PR DESCRIPTION
**Pull request by unicodeveloper**

### CODE REVIEW

File: protected-routes.js [hunk 1]
```diff
@@ -5,11 +5,23 @@ var express = require('express'),
...
 var jwtCheck = jwt({
-  secret: config.secret
+  secret: config.secret,
+  audience: config.audience,
+  issuer: config.issuer
 });
 
-app.use('/api/protected', jwtCheck);
+// Check for scope
+function require_scope(scope) {
```
Comment (sebadoom) [inline protected-routes.js]: I would use camelCase for function names as is the standard JS convention.

File: protected-routes.js [hunk 2]
```diff
@@ -5,11 +5,23 @@ var express = require('express'),
...
+  return function (req, res, next) {
+    var has_scopes = req.user.scope === scope;
+    if (!has_scopes) { res.send(401); }
```
Comment (sebadoom) [inline protected-routes.js]: Function should return after 401, otherwise next() is called (which you don't want).

File: user-routes.js [hunk 3]
```diff
@@ -83,6 +107,7 @@ app.post('/sessions/create', function(req, res) {
   }
 
   res.status(201).send({
-    id_token: createToken(user)
+    id_token: createToken(user),
```
Comment (sebadoom) [inline user-routes.js]: I'd rename createToken to createIdToken

Comment (unicodeveloper) [reply user-routes.js]: Cool

File: protected-routes.js (hunk 2 above)
Comment (unicodeveloper) [reply protected-routes.js]: @sebadoom I'm not clear about this. Can you elaborate?

File: protected-routes.js (hunk 1 above)
Comment (unicodeveloper) [reply protected-routes.js]: Will do that now

File: protected-routes.js (hunk 2 above)
Comment (sebadoom) [reply protected-routes.js]: This should be:

```
if (!has_scopes) { 
  res.send(401); 
  return;
}
```

Comment (unicodeveloper) [reply protected-routes.js]: Thanks @sebadoom

**Pull request merged**